### PR TITLE
Dispose CancellationTokenSource

### PR DESCRIPTION
**Pull request by Youssef1313**

### CODE REVIEW

File: src/Cli/dotnet/NugetSearch/NugetToolSearchApiRequest.cs [hunk 1]
```diff
@@ -23,8 +23,8 @@ public async Task<string> GetResult(NugetSearchApiParameter nugetSearchApiParame
                 nugetSearchApiParameter.Prerelease);
 
             var httpClient = new HttpClient();
-            var cancellation = new CancellationTokenSource(TimeSpan.FromSeconds(10));
-            HttpResponseMessage response = await httpClient.GetAsync(queryUrl, cancellation.Token);
+            using var cancellation = new CancellationTokenSource(TimeSpan.FromSeconds(10));
+            HttpResponseMessage response = await httpClient.GetAsync(queryUrl, cancellation.Token).ConfigureAwait(false);
```
Comment (sharwell) [inline src/Cli/dotnet/NugetSearch/NugetToolSearchApiRequest.cs:27]: 💭 It does not appear that this file uses explicit `ConfigureAwait` calls.

Comment (Youssef1313) [reply src/Cli/dotnet/NugetSearch/NugetToolSearchApiRequest.cs:27]: After a second review, I even think it's not recommended to use ConfigureAwait(false) here.
Thanks for review, I'll revert it.

**Review by wli3 — APPROVED** [review]

**Comment by wli3** [comment]
@sharwell since you are here. Is there any analyzer to watch out for disposing CancellationTokenSource ?

**Comment by Youssef1313** [comment]
@wli3 I found this
https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2000

**Pull request merged**